src/hotspot/os_cpu/linux_arm/javaThread_linux_arm.cpp

```diff
@@ -40,8 +40,13 @@ frame JavaThread::pd_last_frame() {
 }
 
 void JavaThread::cache_global_variables() {
+#if INCLUDE_SHENANDOAHGC
```

Comment:
Sounds like we want to be consistent between C1 and C2 code, so maybe we should inject in adjacent block as:

```
 if (bs->is_a(BarrierSet::CardTableBarrierSet) &&
     !bs->is_a(BarrierSet::ShenandoahBarrierSet)) {
  ...
```